structure/stack/linked_list_test.go

```diff
@@ -0,0 +1,51 @@
+package stack
+
+import "testing"
+
+// TestStackLinkedList for testing stack implementation using singly linked list
+func TestStack_SinglyLinkedList(t *testing.T) {
+	st := NewLinkedList[int]()
+
+	st.Push(1)
+	st.Push(2)
+
+	t.Run("Stack Push", func(t *testing.T) {
+		result := st.ToSlice()
+		expected := []int{2, 1}
+		for x := range result {
+			if result[x] != expected[x] {
+				t.Errorf("Expected stack elements to be %v. Current elements: %v", expected, result)
+			}
+		}
+	})
+
+	t.Run("Stack isEmpty", func(t *testing.T) {
+		if st.Empty() {
+			t.Error("Stack shouldn't be emtpy")
+		}
+	})
+
+	t.Run("Stack Length", func(t *testing.T) {
+		if st.Len() != 2 {
+			t.Errorf("Expected stack length to be 2, got %d", st.Len())
+		}
+	})
+
+	st.Pop()
```

Comment:
This needs to be checked. Please fix all the listing and styling errors raised by the CI.